### ISSUE TITLE
wrong probability prediction after loading model from disk

### ISSUE THREAD

**Issue by yairdata** [microsoft/LightGBM#3300]
i have created a lightGBM model , saved it as txt file with save_model method.
then i have created pmml from it and scored my test set.
in parallel i loaded back the model and scored the test set.
the predicted probabilities where different.
looking at the predictions themselves  - seems like the pmml scores are correct and the python ones are not.
important note - a lot of the python scores were the same , while in pmml it was a lot more sparse.
the python code is :

`lgbModel2 = lgb.Booster(model_file=path,params=mydict,silent=False)`
`df['probability'] = lgbModel2.predict(df[categorical_columns+numeric_columns])`

mydict is a dictionary i created from the model.txt file manually (the params were not loaded without doing so out of the box)
i have heavy suspicion that the model is not fully loaded back correctly , and that is the reason for the incompatibility .
using the python version , on CPU .

**Comment by guolinke** [comment]
Did you check the version of LightGBM used in the training model and python prediction?
Did you ensure the feature order in `df[categorical_columns+numeric_columns]` are the same as in training?
Did you check the output transformation in prediction, like sigmoid in binary classification?

sorry, I don't quite understand your process you encounter the problem. 
It will be better if you can reproduce the problem with some random data, and your model file.
